server/api/db/sqldb/db.py

```diff
@@ -2925,6 +2926,21 @@ def _update_feature_set_features(
                 )
                 update_labels(feature, labels)
                 feature_set.features.append(feature)
+            elif feature_name not in features_to_remove:
+                # check if the existing feature's labels were changed, and update them if needed
+                feature = next(
+                    (
+                        feature
+                        for feature in feature_set.features
+                        if feature.name == feature_name
+                    ),
+                    None,
+                )
+                labels = feature_dict.get("labels") or {}
+                if not server.api.utils.helpers.are_object_labels_equal_to_labels_dict(
+                    feature.labels, labels
+                ):
```

Comment:
I don't see why you need this check. Just update the labels

Reply:
Actually yeah I agree with Alon on this one

Reply:
but why update if there is nothing to update?

Reply:
It's the same runtime and memory usage as checking first and then not updating

Reply:
alright then, I'll remove the entire extra logic :)